web/frontend/src/components/slam-2d-render.vue

```diff
@@ -39,9 +40,40 @@ const props = defineProps<{
   resources: commonApi.ResourceName.AsObject[]
   pointcloud?: Uint8Array
   pose?: commonApi.Pose
+  destExists?: boolean
+  destVector?: THREE.Vector3
+  axes?: boolean
 }
 >();
 
+const baseUrl = `data:image/svg+xml,%3Csvg width='31' height='39' viewBox='0 0 31 39' fill='none' 
+                xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M3.55907 4.21112L1.70584 2.78033L1.95357
+                5.10847L4.71496 31.0594L5.08708 34.5565L6.61063 31.3869L11.7355 20.7249L23.5021 21.9458L27.0001
+                22.3088L24.2164 20.1596L3.55907 4.21112Z' fill='%23BE3536' stroke='%23FCECEA'
+                stroke-width='2'/%3E%3C/svg%3E`;
+                
+const destUrl = `data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' 
+                xmlns='http://www.w3.org/2000/svg'%3E%3Cg clip-path='url(%23clip0_2995_26851)'%3E%3Cpath
+                d='M12 22L11.2579 22.6703L12 23.4919L12.7421 22.6703L12 22ZM12 22C12.7421 22.6703 12.7422 22.6701 
+                12.7424 22.67L12.7429 22.6694L12.7443 22.6679L12.749 22.6627L12.7658 22.6439C12.7803 22.6277 
+                12.8012 22.6042 12.8281 22.5737C12.8819 22.5127 12.9599 22.4238 13.0584 22.3094C13.2554 22.0808 
+                13.5352 21.75 13.8702 21.3372C14.5393 20.5127 15.4332 19.3558 16.329 18.0281C17.2228 16.7032 
+                18.1311 15.1902 18.8189 13.6549C19.5008 12.1328 20 10.5148 20 9C20 4.57772 16.4223 1 12 1C7.57772 
+                1 4 4.57772 4 9C4 10.5148 4.49923 12.1328 5.18115 13.6549C5.86894 15.1902 6.77715 16.7032 7.67104 
+                18.0281C8.56684 19.3558 9.46066 20.5127 10.1298 21.3372C10.4648 21.75 10.7446 22.0808 10.9416 
+                22.3094C11.0401 22.4238 11.1181 22.5127 11.1719 22.5737C11.1988 22.6042 11.2197 22.6277 11.2342 
+                22.6439L11.251 22.6627L11.2557 22.6679L11.2571 22.6694L11.2576 22.67C11.2578 22.6701 11.2579 
+                22.6703 12 22ZM8 9C8 6.79228 9.79228 5 12 5C14.2077 5 16 6.79228 16 9C16 10.169 15.3927 11.7756 
+                14.4208 13.5258C13.7032 14.818 12.8343 16.106 12.004 17.2276C11.1827 16.1085 10.3144 14.8168 
+                9.59393 13.5214C8.61458 11.7606 8 10.1512 8 9Z' fill='%23BE3536' stroke='%23FCECEA' 
+                stroke-width='2'/%3E%3Cpath d='M12 12.5C13.933 12.5 15.5 10.933 15.5 9C15.5 7.067 13.933 5.5 12 
+                5.5C10.067 5.5 8.5 7.067 8.5 9C8.5 10.933 10.067 12.5 12 12.5Z' fill='%23BE3536' 
+                stroke='%23FCECEA' stroke-width='2'/%3E%3C/g%3E%3Cdefs%3E%3CclipPath 
+                id='clip0_2995_26851'%3E%3Crect width='24' height='24' 
+                fill='white'/%3E%3C/clipPath%3E%3C/defs%3E%3C/svg%3E`;
+
+const emit = defineEmits<{(event: 'click', point: THREE.Vector3): void}>();
```

Comment:
What does this do?

Reply:
this allows up the track a clicked point on the 2d slam map;